src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1987,27 +1974,25 @@ private IExpressionStatementOperation CreateBoundExpressionStatementOperation(Bo
             return new ExpressionStatementOperation(expression, _semanticModel, syntax, isImplicit);
         }
 
-        internal IOperation CreateBoundTupleLiteralOperation(BoundTupleLiteral boundTupleLiteral, bool createDeclaration = true)
-        {
-            return CreateTupleOperation(boundTupleLiteral, boundTupleLiteral.Type, createDeclaration);
-        }
-
-        internal IOperation CreateBoundConvertedTupleLiteralOperation(BoundConvertedTupleLiteral boundConvertedTupleLiteral, bool createDeclaration = true)
-        {
-            return CreateTupleOperation(boundConvertedTupleLiteral, boundConvertedTupleLiteral.SourceTuple?.Type, createDeclaration);
-        }
-
-        internal IOperation CreateTupleOperation(BoundTupleExpression boundTupleExpression, TypeSymbol? naturalType, bool createDeclaration)
+        internal IOperation CreateBoundTupleOperation(BoundTupleExpression boundTupleExpression, bool createDeclaration = true)
         {
             SyntaxNode syntax = boundTupleExpression.Syntax;
             bool isImplicit = boundTupleExpression.WasCompilerGenerated;
             ITypeSymbol? type = boundTupleExpression.GetPublicTypeSymbol();
+            TypeSymbol? naturalType = boundTupleExpression switch
```

Comment:
>TypeSymbol? naturalType [](start = 12, length = 23)

Consider moving this below the `if (syntax is DeclarationExpressionSyntax) { ... }` block.